### PR TITLE
feat(movie api): Route, Controller and Tests for Movie API

### PR DESCRIPTION
**Pull request by mpiercy827**
What: Created route, controller and tests for a simple movie API.

Why: Need to allow movies to be created through the API.

Details:
- Implemented route for creating movies (POST to /movies).
- Validate payload with Joi.
- Added unit tests as well as integrated tests.

### CODE REVIEW

File: lib/plugins/features/movies/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const Controller = require('./controller');
+const MovieValidator = require('../../../validators/movie');
```
Comment (charlesbjohnson) [inline lib/plugins/features/movies/index.js]: nit: [align `=`](https://github.com/lob/wiki/blob/master/engineering/style-guides/node.md#align-equal-signs-of-all-requires-within-each-block)

File: test/validators/movie.test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    it('is less than 255 characters', () => {
+      const payload = { title: 'Title'.repeat(60) };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path).to.eql('title');
+      expect(result.error.details[0].type).to.eql('string.max');
+    });
+
+  });
```
Comment (charlesbjohnson) [inline test/validators/movie.test.js]: Does this validator also allow empty titles? We might want to enforce that the title consists of at least one character.

**Comment by charlesbjohnson** [comment]
LGTM pending 1

File: test/validators/movie.test.js (hunk 2 above)
Comment (mpiercy827) [inline test/validators/movie.test.js]: Added a check to make sure the title was nonempty!

**Comment by mgartner** [comment]
LGMT 💯

**Pull request merged**